src/js/components/DataTable/index.d.ts

```diff
@@ -98,7 +98,14 @@ export interface DataTableProps<TRowType = any> {
       pad?: PadType;
     };
   };
-  rowDetails?: (row: TRowType) => React.ReactNode;
+  rowDetails?:
+    | ((row: TRowType) => React.ReactNode)
+    | {
+        render: (row: TRowType) => React.ReactNode;
+        expanded: (row: TRowType) => boolean;
+        expandable: (row: TRowType) => boolean;
+        onClickExpand: (row: TRowType) => void;
```

Comment:
`expanded`, `expandable`, and `onClickExpand` should all be optional. Also in [this comment](https://github.com/grommet/grommet/pull/6571#issuecomment-1563442478) @taysea mentioned changing `expanded` and `expandable` to be arrays of string or numbers where the items in the arrays are primary keys of rows.